Packs/BastilleNetworks/Integrations/BastilleNetworks/BastilleNetworks.py

```diff
@@ -0,0 +1,372 @@
+
+    readable_fields = ['event_id', 'area', 'tags', 'device_info', 'emitter',
+                       'first_seen', 'last_seen']
+
+    raw_response = copy.deepcopy(events)
+    context_events, readable_events = parse_events(events, readable_fields)
+
+    readable_output = tableToMarkdown('Device Events', readable_events, readable_fields)
+    context_output = {
+        'Bastille.DeviceEvent(val.event_id == obj.event_id)': context_events}
+
+    return readable_output, context_output, raw_response
+
+
+def add_device_tag_command(client, args):
+    """ Command to add tag to an existing admin devices entry """
+
+    site, concentrator, map = get_site_params()
+
+    transmitter_id = args.get('transmitter_id')
+    tag = args.get('tag')
+
+    result = client.add_device_tag(site=site, concentrator=concentrator, map=map,
+                                   transmitter_id=transmitter_id, tag=tag)
+
+    if 'status' not in result:
+        return_error('Failed to add device tag')
+
+    return result['status'], {}, result
+
+
+def remove_device_tag_command(client, args):
+    """ Command to remove tag from an existing admin devices entry """
+
+    site, concentrator, map = get_site_params()
+
+    transmitter_id = args.get('transmitter_id')
+    tag = args.get('tag')
+
+    result = client.remove_device_tag(site=site, concentrator=concentrator, map=map,
+                                      transmitter_id=transmitter_id, tag=tag)
+
+    if 'status' not in result:
+        return_error('Failed to remove device tag')
+
+    return result['status'], {}, result
+
+
+def fetch_incidents(client, last_run, first_fetch_time):
+    """ Callback to fetch incidents periodically """
+
+    last_fetch_time = last_run.get('last_fetch', first_fetch_time)
+
+    site, concentrator, map = get_site_params()
+
+    params = demisto.params()
+    tags = params.get('tags') or None
+    event_types = params.get('event_types')
+
+    zone_events = []
+    if 'all' in event_types or 'zone_event' in event_types:
+        zone_events = client.get_zone_events(concentrator=concentrator, map=map,
+                                             site=site, tags=tags, since=last_fetch_time)
+
+    device_events = []
+    if 'all' in event_types or 'device_event' in event_types:
+        device_events = client.get_device_events(concentrator=concentrator, map=map,
+                                                 site=site, tags=tags,
+                                                 since=last_fetch_time)
+
+    events = zone_events + device_events
+    current_fetch_time = last_fetch_time
+
+    incidents = []
+    for event in events:
+        event_time = int(event['time_s'])
+        incident = {
+            'name': event['event_type'],
+            'occurred': datetime.utcfromtimestamp(event_time).strftime(
+                DATE_FORMAT),
+            'rawJSON': json.dumps(event),
+        }
+        incidents.append(incident)
```

Comment:
```suggestion
       if event_time > current_fetch_time:
             incidents.append(incident)
```

This should only append in case the current event happened after the last fetch.